### PR TITLE
[CN 444] Add Executor Service e2e test

### PR DESCRIPTION
**Pull request by cagric0**

### CODE REVIEW

File: bundle.yaml [hunk 1]
```diff
@@ -2994,6 +2994,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.name
+        - name: DEVELOPER_MODE_ENABLED
+          value: "true"
```
Comment (SeriyBg) [inline bundle.yaml:2998]: Leftover?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -9,6 +9,9 @@ images:
   newName: hazelcast/hazelcast-platform-operator
   newTag: latest-snapshot
 patches:
+- path: enable_developer_mode.yaml
+  target:
+    kind: Deployment
```
Comment (SeriyBg) [inline config/manager/kustomization.yaml:14]: Same here?

File: controllers/hazelcast/hazelcast_resources.go [hunk 3]
```diff
@@ -434,12 +434,15 @@ func hazelcastConfigMapData(ctx context.Context, c client.Client, h *hazelcastv1
...
 
 	cfg := hazelcastConfigMapStruct(h)
+
 	err = fillHazelcastConfigWithMaps(ctx, c, &cfg, h, ml)
 	if err != nil {
 		return nil, err
 	}
 	fillHazelcastConfigWithExecutorServices(&cfg, h)
 
+	fillHazelcastConfigWithExecutorServices(&cfg, h)
```
Comment (SeriyBg) [inline controllers/hazelcast/hazelcast_resources.go:444]: Why is this method called twice?

File: test/e2e/helpers_test.go [hunk 4]
```diff
@@ -483,3 +519,44 @@ func printDebugState() {
...
+	}
+	for i, des1 := range es1["des"].([]hazelcastcomv1alpha1.DurableExecutorServiceConfiguration) {
+		assertDurableES(des1, es2.Durable[i])
+	}
+	for i, ses1 := range es1["ses"].([]hazelcastcomv1alpha1.ScheduledExecutorServiceConfiguration) {
+		assertScheduledES(ses1, es2.Scheduled[i])
+	}
+}
+
+func assertBasicES(bes1 hazelcastcomv1alpha1.ExecutorServiceConfiguration, bes2 codecTypes.ExecutorServiceConfig) {
```
Comment (SeriyBg) [inline test/e2e/helpers_test.go:543]: I would suggest renaming the parameters to something like `expected` and `actual`, as the `1/2` suffix makes it kinda confusing

**Review by SeriyBg — CHANGES_REQUESTED** [review]

File: bundle.yaml (hunk 1 above)
Comment (cagric0) [reply bundle.yaml:2998]: Yes, thank you!

File: test/e2e/hazelcast_user_code_test.go [hunk 5]
```diff
@@ -116,4 +116,71 @@ var _ = Describe("Hazelcast Map Config With User Code Deployment", Label("map"),
...
+		stopChan, readyChan := portForwardPod(hazelcast.Name+"-0", hazelcast.Namespace, localPort+":5701")
+		defer closeChannel(stopChan)
+		err := waitForReadyChannel(readyChan, 5*Second)
+		Expect(err).To(BeNil())
+
+		By("checking if the initially added executor service configs are created correctly")
+		cl := createHazelcastClient(context.Background(), hazelcast, localPort)
+		defer func() {
+			err = cl.Shutdown(context.Background())
+			Expect(err).To(BeNil())
```
Comment (SeriyBg) [inline test/e2e/hazelcast_user_code_test.go:156]: Small nit
```suggestion
			Expect(cl.Shutdown(context.Background())).Should(Succeed())
```

File: test/e2e/hazelcast_user_code_test.go [hunk 6]
```diff
@@ -85,6 +86,10 @@ var _ = Describe("Hazelcast Map Config With User Code Deployment", Label("map"),
 		By("Filling the map with entries")
 		entryCount := 5
 		cl := createHazelcastClient(context.Background(), hazelcast, localPort)
+		defer func() {
+			err = cl.Shutdown(context.Background())
+			Expect(err).To(BeNil())
```
Comment (SeriyBg) [inline test/e2e/hazelcast_user_code_test.go:91]:
```suggestion
			Expect(cl.Shutdown(context.Background())).Should(Succeed())
```

**Review by dzeromski-hazelcast — APPROVED** [review]

**Review by SeriyBg — APPROVED** [review]
Approved with small nits 😄

**Pull request merged**